projects/Buttons_and_ADC/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+static void prv_button_interrupt_handler(const GpioAddress *adc_address, void *context) {
+  GpioAddress *adc_position = context;
+  uint16_t adc = 0;
```

Comment:
This should be named more descriptively. What is `adc`? Is it the data, the channel, the address? Right now its too ambigious